### PR TITLE
Transform for F# Pipeline

### LINKED ISSUE #6889: [pipeline operator] support await
Choose one: is this a bug report or feature request?

### Input Code

```js
promise |> await
```

which is the equivalent of

```js
await promise
```

### Expected Behavior

```js
var _promise;

_promise = promise, await _promise;
```

### Current Behavior

```
Parse error: await is a reserved word (1:11)
```

### Your Environment

| software         | version(s)
| ---------------- | -------
| Babel            |  
| Babylon          | 
| node             | 
| npm              | 
| Operating System | 

As described in the spec [here](https://github.com/tc39/proposal-pipeline-operator#use-of-await).

### PR DESCRIPTION
**Pull request by thiagoarrais**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #6889 
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      | Yes
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

This adds the transform for the `fsharp` pipeline proposal.

It is a port to the main babel repo of the work that happened in https://github.com/mAAdhaTTah/babel/pull/1

cc @nicolo-ribaudo @mAAdhaTTah

### CODE REVIEW

File: packages/babel-plugin-proposal-pipeline-operator/test/fixtures/minimal/optimize-0-param-arrow/output.js [hunk 1]
```diff
@@ -1,3 +1,5 @@
+var _ref, _a;
```
Comment (nicolo-ribaudo) [inline packages/babel-plugin-proposal-pipeline-operator/test/fixtures/minimal/optimize-0-param-arrow/output.js]: It seems that these aren't needed :thinking:

File: packages/babel-plugin-proposal-pipeline-operator/src/buildOptimizedSequenceExpression.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import { types as t } from "@babel/core";
+
+// tries to optimize sequence expressions in the format
+//   (a = b, ((c) => d + e)(a))
+// to
+//   (a = b, a + e)
+const buildOptimizedSequenceExpression = ({ assign, call, path }) => {
```
Comment (nicolo-ribaudo) [inline packages/babel-plugin-proposal-pipeline-operator/src/buildOptimizedSequenceExpression.js]: (Note for reviewers: this function isn't new; it has been moved here from https://github.com/babel/babel/blob/b5febcf1f993cc3ec36fda7c7908181048a0069f/packages/babel-plugin-proposal-pipeline-operator/src/minimalVisitor.js)

File: packages/babel-plugin-proposal-pipeline-operator/test/fixtures/minimal/optimize-0-param-arrow/output.js (hunk 1 above)
Comment (thiagoarrais) [reply packages/babel-plugin-proposal-pipeline-operator/test/fixtures/minimal/optimize-0-param-arrow/output.js]: Nope... I'm almost sure that they're an artifact of declaring the placeholder in the visitor and then throwing it away during optimization.

Comment (thiagoarrais) [reply packages/babel-plugin-proposal-pipeline-operator/test/fixtures/minimal/optimize-0-param-arrow/output.js]: Found a quick solution in 29fa385. Let me know what you think.

**Comment by mAAdhaTTah** [comment]
🎉

**Comment by thiagoarrais** [comment]
🎉

**Pull request merged**